cylc/flow/task_pool.py

```diff
@@ -478,6 +479,7 @@ def release_runahead_task(self, itask):
         self.pool.setdefault(itask.point, {})
         self.pool[itask.point][itask.identity] = itask
         self.pool_changed = True
+        self.pool_changes.append(itask)
```

Comment:
I don't track removed tasks, as I prune by cycle  point to allow for a full graph in each (for N-edges query data).. And they are still used with the current `get_all_tasks` ...etc:
```
    def get_tasks(self):
        """Return a list of task proxies in the main task pool."""
        if self.pool_changed:
            self.pool_changed = False
            self.pool_list = []
            for itask_id_maps in self.queues.values():
                self.pool_list.extend(list(itask_id_maps.values()))
        return self.pool_list

    def get_rh_tasks(self):
        """Return a list of task proxies in the runahead pool."""
        if self.rhpool_changed:
            self.rhpool_changed = False
            self.rhpool_list = []
            for itask_id_maps in self.runahead_pool.values():
                self.rhpool_list.extend(list(itask_id_maps.values()))
        return self.rhpool_list
```
Maybe once we migrate to only using the data store (GraphQL CLI), then we can revisit.

Reply:
This could still be fixed:
> Note: it was initialised as a list, but set to a boolean in the line above.

```sh
(venv) [oliverh@drugs-and-money cylc-flow]$ git grep 'pool_changed ='

cylc/flow/task_pool.py:        self.pool_changed = []     # <----------------
cylc/flow/task_pool.py:        self.rhpool_changed = []    # <----------------
cylc/flow/task_pool.py:        self.rhpool_changed = True
cylc/flow/task_pool.py:        self.pool_changed = True
cylc/flow/task_pool.py:        self.rhpool_changed = True
cylc/flow/task_pool.py:            self.rhpool_changed = True
cylc/flow/task_pool.py:        self.pool_changed = True
cylc/flow/task_pool.py:            self.pool_changed = False
cylc/flow/task_pool.py:            self.rhpool_changed = False
```

Reply:
Yes not sure why they were like that.. Should that come under this PR? I didn't touch or use them here.

Reply:
If you don't mind slipping those in. (Tiny refactor related to this PR in a sense.)